app/src/openxr/cpp/DeviceDelegateOpenXR.cpp

```diff
@@ -1389,23 +1382,20 @@ DeviceDelegateOpenXR::usesPassthroughCompositorLayer() const {
 void
 DeviceDelegateOpenXR::DeleteLayer(const VRLayerPtr& aLayer) {
   if (m.cubeLayer && m.cubeLayer->layer == aLayer) {
-    m.cubeLayer->Destroy();
-    m.cubeLayer = nullptr;
+    m.cubeLayer.reset();
     return;
   }
   if (m.equirectLayer && m.equirectLayer->layer == aLayer) {
-    m.equirectLayer->Destroy();
-    m.equirectLayer = nullptr;
+    m.equirectLayer.reset();
     return;
   }
   if (m.passthroughLayer && m.passthroughLayer->layer == aLayer) {
-    m.passthroughLayer->Destroy();
-    m.passthroughLayer = nullptr;
+    m.passthroughLayer.reset();
     return;
   }
   for (int i = 0; i < m.uiLayers.size(); ++i) {
     if (m.uiLayers[i]->GetLayer() == aLayer) {
-      m.uiLayers[i]->Destroy();
+      m.uiLayers[i].reset();
       m.uiLayers.erase(m.uiLayers.begin() + i);
```

Comment:
Here it should be enough to just erase the pointer from the array. It'll automatically free the resources as it's the only reference